examples/gatekeeper/imperative/README.md

```diff
@@ -0,0 +1,73 @@
+# gatekeeper: Imperative Example
+
+### Overview
+
+This example demonstrates how to imperatively invoke the [gatekeeper] function
+to validate resources using gatekeeper constraints.
+
+There are 2 resources in `policy.yaml`:
+
+- `ConstraintTemplate`
+- `K8sBannedConfigMapKeysV1`
+
+There is one `ConfigMap` in `config-map.yaml`.
+
+The constraint disallows using `private_key` as a key in the `ConfigMap`.
+
+### Function invocation
+
+Get the package:
+
+```shell
+$ kpt pkg get https://github.com/GoogleContainerTools/kpt-functions-catalog.git/examples/gatekeeper/imperative
+```
+
+Run the function with `--results-dir` flag:
+
+```shell
+kpt fn eval --image=gcr.io/kpt-fn/gatekeeper:unstable --results-dir=results
```

Comment:
> why are we using unstable version here ?

We use `unstable` in master branch.
A released version tag will be used in release branches. We have a CI check for it.

Reply:
> Is there a reason for not displaying the CLI output here ? because the policy violations will be displayed very nicely here.

@frankfarzan and I have a discussion about this before. CLI output is from kpt's representation layer. It may change and our doc will become out-of-date. It's safer to use structured results which is a stable API.
@droot WDYT?

Reply:
> A released version tag will be used in release branches. We have a CI check for it.

ok.

> CLI output is from kpt's representation layer. It may change and our doc will become out-of-date. It's safer to use structured results which is a stable API.

This is the biggest pain points with the docs. May be once `v1.0` is stable, output is less likely to change often, we can revisit this. 

Actually the perfect solution will what `Go` did with the examples with a capability to run it on demand and see the output in in the browser. It will be an interesting project to try it out once our release is out.